src/libraries/Microsoft.PowerFx.Connectors/Texl/ConnectorTexlFunction.cs

```diff
@@ -85,8 +86,11 @@ public override bool TryGetParamDescription(string paramName, out string paramDe
 
         public override bool HasSuggestionsForParam(int argumentIndex) => argumentIndex <= MaxArity;
 
-        public async Task<FormulaValue> InvokeAsync(FormulaValue[] args, IServiceProvider serviceProvider, CancellationToken cancellationToken)
+        public async Task<FormulaValue> InvokeAsync(FunctionInvokeInfo invokeInfo, CancellationToken cancellationToken)
         {
+            FormulaValue[] args = invokeInfo.Args.ToArray(); // $$$ remove ToArray
```

Comment:
>// $$$ remove ToArray [](http://example.com/codeflow?start=61&length=21)

IReadOnlyList vs. Array. 
IRL ensures immutablility. 

We just need to plumb IRL through a bunch of files.